### PR TITLE
client: remove deprecated authenticatedUser.email

### PR DESCRIPTION
**Pull request by bobheadxi**
The `authenticatedUsers.email` field is deprecated - the equivalent can now be found with `authenticatedUsers.emails.find(email => email.isPrimary)`. This gives more details about the email as well, such as whether it is verified or not, which is required to implement verified email checks (e.g. https://github.com/sourcegraph/sourcegraph/pull/46184)

https://github.com/sourcegraph/sourcegraph/blob/fb5a412a9c66c4e08a2fb4dab859a765cdedc1ff/cmd/frontend/graphqlbackend/schema.graphql#L5538-L5542

## Test plan

Tests pass still

## App preview:

- [Web](https://sg-web-01-05-client-remove-deprecated.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #46183** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/46183" target="_blank">Graphite</a>  👈
    * **PR #46184** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/46184" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/46183?utm_source=stack-comment).

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.04 kb) | 0.00% (+0.15 kb) | 0.00% (+0.11 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/01-05-client_remove_deprecated_authenticatedUser.email/compare-report.html;tab=live_object) for a full comparison between the commits 6b80bb1 and 7a6dc29 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by vdavid — APPROVED** [review]
LGTM! If the backend is indeed in place then this should work right. Left two very small nits.

**Review by vovakulikov — APPROVED** [review]

File: client/web/src/Layout.tsx [hunk 1]
```diff
@@ -201,7 +201,14 @@ export const Layout: React.FunctionComponent<React.PropsWithChildren<LayoutProps
                     onSubmit={handleSubmitFeedback}
                     modal={true}
                     openByDefault={true}
-                    authenticatedUser={props.authenticatedUser}
+                    authenticatedUser={
+                        props.authenticatedUser
+                            ? {
+                                  username: props.authenticatedUser?.username || '',
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply client/web/src/Layout.tsx:207]: thank you!

**Review by limitedmage — APPROVED** [review]

**Pull request merged**